test/zdtm/lib/unix.c

```diff
@@ -0,0 +1,25 @@
+#include <sys/socket.h>
+#include <sys/un.h>
+#include "zdtmtst.h"
+#include "fs.h"
+
+int unix_fill_sock_name(struct sockaddr_un *name, char *relFilename)
+{
+	char *cwd;
+
+	if (get_cwd_check_perm(&cwd)) {
+		pr_err("failed to get current working directory with valid permissions.\n");
+		return -1;
+	}
+
+	if (strlen(relFilename) + strlen(cwd) + 1 >= sizeof(name->sun_path)) {
+		pr_err("filename \"%s/%s\" is too long for sockaddr_un initialiazation.\n", 
+			cwd, relFilename);
+		return -1;
+	}
```

Comment:
ssprintf handles path length check itself, and calls abort() to create a coredump so it would be easier to investigate the problem.

Reply:
Agreed